lib/sendstate.c

```diff
@@ -98,17 +98,21 @@ static int check_amount_of_state(quicly_sendstate_t *state)
 {
     size_t num_ranges = state->acked.num_ranges + state->pending.num_ranges;
 
-    /* bail out if number of gaps are small */
-    if (PTLS_LIKELY(num_ranges < 32))
+    /* Bail out if number of gaps are small.
+     * In case of HTTP/3, the worst case is when each HTTP request is received as a separate QUIC packet, and sending a small STREAM
+     * frame carrying a HPACK encoder / decoder in response. If half of those STREAM frames are lost (note: loss of every other
+     * packet can happen during slow start), `num_ranges` can become as large as `request_concurrency * 2`, as each gaps will be
+     * recognized in `acked.num_ranges` and `pending.num_ranges`. */
+    if (PTLS_LIKELY(num_ranges < 256))
```

Comment:
I'm not sure I follow where 256 comes from. Can you explain the number in the comment as well?